09_json/n0npax/pkg/puppy/store/memStore.go

```diff
@@ -0,0 +1,69 @@
+package store
+
+import (
+	"sync"
+
+	puppy "github.com/anz-bank/go-course/09_json/n0npax/pkg/puppy"
+)
+
+// MemStore map based type for storing puppies data
+type MemStore struct {
+	m      map[int]puppy.Puppy
+	nextID int
+	sync.Mutex
+}
+
+// NewMemStore creates new storer for map
+func NewMemStore() *MemStore {
+	return &MemStore{m: map[int]puppy.Puppy{}}
+}
+
+// CreatePuppy creates puppy
+func (m *MemStore) CreatePuppy(p *puppy.Puppy) (int, error) {
+	if p.Value < 0 {
+		return -1, puppy.Errorf(puppy.ErrCodeInvalidInput, "Puppy value have to be positive number")
```

Comment:
"Puppy value has to be equal or bigger than zero", as zero itself is nor positive nor negative (and I'm really running out of nits here)

Reply:
?? This is returning error if p.Value is lower than 0. `0 < 0` -> False so we're jumping to line 26. If `p.Value` is equal or larger than 0 I'm not throwing error? Am I missing sth?

Reply:
0 is positive as far as i remember.

Reply:
Just a nit about the message itself - zero is neither negative nor positive, it's zero. So saying that value has to be positive is not strictly true, because the smallest possible value needs to be zero. So, the message must say `value has to be bigger or equal than zero` or sth else that includes the possibility of a zero value. Nit, really... :)